src/main/java/com/metamx/common/io/smoosh/FileSmoosher.java

```diff
@@ -202,21 +216,116 @@ public boolean isOpen()
       public void close() throws IOException
       {
         open = false;
+        useDelegateSmooshWriter = false;
         internalFiles.put(name, new Metadata(currOut.getFileNum(), startOffset, currOut.getCurrOffset()));
 
         if (bytesWritten != currOut.getCurrOffset() - startOffset) {
-          throw new ISE("WTF? Perhaps there is some concurrent modification going on?");
-        }
-        if (bytesWritten != size) {
-          throw new IOException(
-              String.format("Expected [%,d] bytes, only saw [%,d], potential corruption?", size, bytesWritten)
-          );
+            throw new ISE("WTF? Perhaps there is some concurrent modification going on?");
+          }
+          if (bytesWritten != size) {
+            throw new IOException(
+                String.format("Expected [%,d] bytes, only saw [%,d], potential corruption?", size, bytesWritten)
+            );
+          }        
+        if(!mergeOnProgress && !useDelegateSmooshWriter)
+        {
+        	mergeWithSmoosher();
         }
+
       }
     };
   }
+  
+  public OutputStream makeOutputStream(String filename) throws IOException
+  {
+    File retFile = createdFiles.get(filename);
+    if (retFile == null) {
+      retFile = File.createTempFile("smoosher", filename);
+      retFile.deleteOnExit();
```

Comment:
If you want to know why, you can use your own time to look at the implementation of what that's doing, when it deletes the file and how it goes about it.  Then report back on why you think this should never be used.